tests/test_end_to_end.py

```diff
@@ -448,3 +451,93 @@ def test_driver_validate_with_overrides_2():
         .build()
     )
     assert dr.execute(["d"], overrides={"b": 1})["d"] == 3
+
+
+def test_driver_extra_inputs_can_be_outputs():
+    """Tests that we can request outputs that not in the graph, but are in the inputs."""
+    dr = (
+        driver.Builder()
+        .with_modules(tests.resources.overrides)
+        .with_adapter(base.DefaultAdapter())
+        .build()
+    )
+    actual = dr.execute(["d", "e"], inputs={"a": 1, "e": 10})
+    assert actual["d"] == 4
+    assert actual["e"] == 10
+    # Checks dataframe use case
+    dr = (
+        driver.Builder()
+        .with_modules(tests.resources.example_module)
+        .with_adapter(base.PandasDataFrameResult())
+        .build()
+    )
+    actual = dr.execute(
+        ["avg_3wk_spend", "e"],
+        inputs={"spend": pd.Series([1, 1, 1, 1, 1]), "e": pd.Series([10, 10, 10, 10, 10])},
+    )
+    pd.testing.assert_frame_equal(
+        actual,
+        pd.DataFrame(
+            {
+                "avg_3wk_spend": pd.Series([math.nan, math.nan, 1.0, 1.0, 1.0], dtype=float),
```

Comment:
could be `pd.NaN` and avoid importing `math`

Reply:
No it doesn't work because I need to say it's a float type for the test comparison to work.

```python
pd.Series([pd.NA, pd.NA, 1.0, 1.0, 1.0], dtype=float)
Traceback (most recent call last):
  File "/Users/stefankrawczyk/.pyenv/versions/knowledge_retrieval-py39/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3508, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-8-baac06950501>", line 1, in <module>
    pd.Series([pd.NA, pd.NA, 1.0, 1.0, 1.0], dtype=float)
  File "/Users/stefankrawczyk/.pyenv/versions/3.9.13/envs/knowledge_retrieval-py39/lib/python3.9/site-packages/pandas/core/series.py", line 470, in __init__
    data = sanitize_array(data, index, dtype, copy)
  File "/Users/stefankrawczyk/.pyenv/versions/3.9.13/envs/knowledge_retrieval-py39/lib/python3.9/site-packages/pandas/core/construction.py", line 622, in sanitize_array
    subarr = _try_cast(data, dtype, copy, raise_cast_failure)
  File "/Users/stefankrawczyk/.pyenv/versions/3.9.13/envs/knowledge_retrieval-py39/lib/python3.9/site-packages/pandas/core/construction.py", line 840, in _try_cast
    subarr = np.array(arr, dtype=dtype, copy=copy)
TypeError: float() argument must be a string or a number, not 'NAType'
```
(there is no NaN)